modules/data-entry/src/main/frontend/src/questionnaireEditor/ReferenceInput.jsx

```diff
@@ -266,16 +281,38 @@ let ReferenceInput = (props) => {
     <EditorInput name={objectKey} hint={hint}>
       <input type="hidden" name={objectKey + "@TypeHint"} value='Reference' />
       {hiddenInput}
-      <Select
-        variant="standard"
-        id={objectKey}
-        value={curValue || []}
-        onChange={(event) => {changeCurValue(event.target.value);}}
-        input={<Input id={objectKey} />}
-        renderValue={(value) => titleMap[value]}
-      >
-        {getReactComponentFromFields(options)}
-      </Select>
+      <Autocomplete
+        value={curValue && autoselectOptions.find(item => item.uuid == curValue) || null}
+        options={autoselectOptions}
+        filterOptions={filterOptions}
+        groupBy={(option) => option.category}
+        onChange={(event, value) => {
+          setCurValue(value?.uuid || "");
```

Comment:
Regression - the stats context enabling variable restrictions is no longer updated.
```suggestion
          changeCurValue(value?.uuid || "");
```